### PR TITLE
Remove usages of ZERO_ON_NEW macro

### PR DESCRIPTION
**Pull request by sbx320**
Requires testing. I don't have a client available right now, so I can't test that part.

### CODE REVIEW

File: Server/mods/deathmatch/logic/CAccount.h [hunk 1]
```diff
@@ -117,20 +116,20 @@ class CAccount
 
     CAccountManager* m_pManager;
 
-    EAccountType              m_AccountType;
+    EAccountType              m_AccountType = EAcountType::Guest;
```
Comment (qaisjp) [inline Server/mods/deathmatch/logic/CAccount.h]:
```suggestion
    EAccountType              m_AccountType = EAccountType::Guest;
```

**Comment by ccw808** [comment]
No

**Comment by qaisjp** [comment]
Next error:

```cpp
In file included from ../Server/mods/deathmatch/StdInc.h:59:
In file included from ../Server/mods/deathmatch/logic/packets/CEntityAddPacket.h:18:
../Server/mods/deathmatch/logic/packets/../CGame.h:531:54: error: no viable conversion from 'nullptr_t' to 'CCommandLineParser'
    CCommandLineParser         m_CommandLineParser = nullptr;
                                                     ^~~~~~~
```

**Comment by Dutchman101** [comment]
There are multiple occurences of ZERO_ON_NEW in PR #760.
If this goes through, then we should ensure that no pending PR's will re-introduce the macro's usage.

File: Client/mods/deathmatch/logic/CClientGame.h [hunk 2]
```diff
@@ -626,187 +625,186 @@ class CClientGame
...
+    bool                      m_bMimicLag = false;
+    unsigned long             m_ulLastMimicLag = 0;
     CVector                   m_vecLastMimicPos;
     CVector                   m_vecLastMimicMove;
     CVector                   m_vecLastMimicTurn;
     CVector                   m_vecLastMimicRot;
-    bool                      m_bDoPaintballs;
-    bool                      m_bShowInterpolation;
+    bool                      m_bDoPaintballs = 0;
+    bool                      m_bShowInterpolation = 0;
```
Comment (StrixG) [inline Client/mods/deathmatch/logic/CClientGame.h:803]:
```suggestion
    bool                      m_bShowInterpolation = false;
```

File: Client/mods/deathmatch/logic/CClientGame.h [hunk 3]
```diff
@@ -626,187 +625,186 @@ class CClientGame
...
     #endif
-    bool m_bDevelopmentMode;
-    bool m_bShowCollision;
-    bool m_bShowSound;
+    bool m_bDevelopmentMode = 0;
+    bool m_bShowCollision = 0;
+    bool m_bShowSound = 0;
```
Comment (StrixG) [inline Client/mods/deathmatch/logic/CClientGame.h:807]:
```suggestion
    bool m_bShowSound = false;
```

File: Client/mods/deathmatch/logic/CClientGame.h [hunk 4]
```diff
@@ -626,187 +625,186 @@ class CClientGame
...
-    bool                      m_bDoPaintballs;
-    bool                      m_bShowInterpolation;
+    bool                      m_bDoPaintballs = 0;
+    bool                      m_bShowInterpolation = 0;
     #endif
-    bool m_bDevelopmentMode;
-    bool m_bShowCollision;
-    bool m_bShowSound;
+    bool m_bDevelopmentMode = 0;
+    bool m_bShowCollision = 0;
```
Comment (StrixG) [inline Client/mods/deathmatch/logic/CClientGame.h:806]:
```suggestion
    bool m_bShowCollision = false;
```

File: Client/mods/deathmatch/logic/CClientGame.h [hunk 5]
```diff
@@ -626,187 +625,186 @@ class CClientGame
...
     CVector                   m_vecLastMimicRot;
-    bool                      m_bDoPaintballs;
-    bool                      m_bShowInterpolation;
+    bool                      m_bDoPaintballs = 0;
+    bool                      m_bShowInterpolation = 0;
     #endif
-    bool m_bDevelopmentMode;
-    bool m_bShowCollision;
-    bool m_bShowSound;
+    bool m_bDevelopmentMode = 0;
```
Comment (StrixG) [inline Client/mods/deathmatch/logic/CClientGame.h:805]:
```suggestion
    bool m_bDevelopmentMode = false;
```

File: Client/mods/deathmatch/logic/CClientGame.h [hunk 6]
```diff
@@ -626,187 +625,186 @@ class CClientGame
...
-    unsigned long             m_ulLastMimicLag;
+    bool                      m_bMimicLag = false;
+    unsigned long             m_ulLastMimicLag = 0;
     CVector                   m_vecLastMimicPos;
     CVector                   m_vecLastMimicMove;
     CVector                   m_vecLastMimicTurn;
     CVector                   m_vecLastMimicRot;
-    bool                      m_bDoPaintballs;
-    bool                      m_bShowInterpolation;
+    bool                      m_bDoPaintballs = 0;
```
Comment (StrixG) [inline Client/mods/deathmatch/logic/CClientGame.h:802]:
```suggestion
    bool                      m_bDoPaintballs = false;
```

File: Server/mods/deathmatch/logic/net/CSimPlayer.h [hunk 7]
```diff
@@ -31,35 +30,35 @@ class CSimPlayer
     NetServerPlayerID&                        GetSocket() { return m_PlayerSocket; };
 
     // General synced vars
-    bool                               m_bIsJoined;
-    unsigned short                     m_usBitStreamVersion;
-    NetServerPlayerID                  m_PlayerSocket;
+    bool                               m_bIsJoined = false;
+    unsigned short                     m_usBitStreamVersion = 0;
+    NetServerPlayerID                  m_PlayerSocket ;
```
Comment (StrixG) [inline Server/mods/deathmatch/logic/net/CSimPlayer.h:35]:
```suggestion
    NetServerPlayerID                  m_PlayerSocket;
```

**Comment by ccw808** [comment]
I think it would be safer if we could remove ZERO_ON_NEW in an automated way

**Comment by sbx320** [comment]
I just found that clang-tidy can apparently automate that for us:

https://clang.llvm.org/extra/clang-tidy/checks/cppcoreguidelines-pro-type-member-init.html

Getting clang-tidy to work with MTA is slightly complicated (as premake has no support for it), but I've done that before. The only downside to this it that we'd end up with everything in the constructor rather than inline in the headers, but I guess that's fine. 

I'll see if I can get this going tomorrow.

**Pull request closed without merging**